sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3178,6 +3178,34 @@ class Dataset[T] private[sql](
     new DataFrameWriter[T](this)
   }
 
+  /**
+   * Create a write configuration builder for v2 sources.
+   *
+   * This builder is used to configure and execute write operations. For example, to append to an
+   * existing table, run:
+   *
+   * {{{
+   *   df.writeTo("catalog.db.table").append()
+   * }}}
+   *
+   * This can also be used to create or replace existing tables:
+   *
+   * {{{
+   *   df.writeTo("catalog.db.table").partitionedBy($"col").createOrReplace()
+   * }}}
+   *
+   * @group basic
+   * @since 3.0.0
+   */
+  def writeTo(table: String): DataFrameWriterV2[T] = {
```

Comment:
We can't change the behavior of `write` because we don't want to break older jobs. And we need to pass the table name or path somewhere. I think this works, but if everyone prefers `writeV2`, we can rename it.

Reply:
I like `writeTo`, since we're explaining exactly what we're writing to at that point.